wp-includes/theme-compat/500.php

```diff
@@ -16,20 +16,10 @@
 ?>
 <main>
 	<h1><?php esc_html_e( 'Oops! Something went wrong.', 'pwa' ); ?></h1>
-	<p><?php esc_html_e( 'Something prevented the page from being rendered. Please try again.', 'pwa' ); ?></p>
-	<details id="error-details" hidden>
+	<?php wp_service_worker_error_message_placeholder(); ?>
+	<details id="error-details">
 		<summary><?php esc_html_e( 'More details', 'pwa' ); ?></summary>
-		<iframe style="width:100%;" srcdoc=""></iframe>
-		<script>
-		function renderErrorDetails( data ) {
-			if ( data.bodyText.trim().length ) {
-				const details = document.getElementById( 'error-details' );
-				details.querySelector( 'iframe' ).srcdoc = data.bodyText;
-				details.hidden = false;
-			}
-		}
-		</script>
-		<?php wp_print_service_worker_error_details_script( 'renderErrorDetails' ); ?>
+		<iframe style="width:100%;" srcdoc="<?php wp_service_worker_error_details_placeholder(); ?>"></iframe>
```

Comment:
Something else to consider is that `srcdoc` is not supported by all browsers. So perhaps it would make more sense to encode the `errorText` as an HTML data URI and then populate the `src` instead.